Purchases/Error Handling/ErrorUtils.swift

```diff
@@ -347,4 +320,44 @@ private extension ErrorUtils {
         return errorWithUserInfo
     }
 
+    private static func logErrorIfNeeded(_ code: ErrorCode) {
+        switch code {
+        case .networkError,
+                .unknownError,
+                .receiptAlreadyInUseError,
+                .unexpectedBackendResponseError,
+                .invalidReceiptError,
+                .invalidAppUserIdError,
+                .operationAlreadyInProgressForProductError,
+                .unknownBackendError,
+                .invalidSubscriberAttributesError,
+                .logOutAnonymousUserError:
+            Logger.error(code.description)
+        case .purchaseCancelledError,
+                .storeProblemError,
+                .purchaseNotAllowedError,
+                .purchaseInvalidError,
+                .productNotAvailableForPurchaseError,
+                .productAlreadyPurchasedError,
+                .missingReceiptFileError,
+                .invalidCredentialsError,
+                .invalidAppleSubscriptionKeyError,
+                .ineligibleError,
+                .insufficientPermissionsError,
+                .paymentPendingError:
+            Logger.appleError(code.description)
+
+        case .receiptInUseByOtherSubscriberError,
+                .configurationError,
+                .unsupportedError,
+                .emptySubscriberAttributes,
+                .productDiscountMissingIdentifierError,
+                .missingAppUserIDForAliasCreationError,
+                .productDiscountMissingSubscriptionGroupIdentifierError:
+            break
+
+        @unknown default:
+            break
```

Comment:
Maybe we still want to log something, pointing out we don't know.

Reply:
👍

Reply:
Actually this was like this already, not all logs were being logged. I didn't realize I should be putting some of these in the right category, then I'll log something in the `default`.